xc/device.ml

```diff
@@ -57,6 +57,22 @@ module Profile = struct
     | x -> debug "unknown device-model profile %s: defaulting to fallback: %s" x (string_of fallback);
        fallback
   let of_domid x = if is_upstream_qemu x then Qemu_upstream else Qemu_trad
+  module Cache = struct
```

Comment:
Is it worth to protect changes to `profiles_of_domids` behind a mutex or there is not and will never be the chance for it to be called concurrently?

Reply:
I think it could happen that we start multiple domains at the same time right?

Reply:
I agree with @mseri. As a global cache, this is prone to losing updates in data races without a mutex.

Reply:
Yes, you are absolutely right, there are threads that could access the data structure concurrently. I've added a mutex protecting access to `profiles_of_domids` in 2fa15127eb7006f4bab2097bca1db6a318242811